sdk/identity/identity/src/credentials/clientCertificateCredential.ts

```diff
@@ -176,7 +180,75 @@ export class ClientCertificateCredential implements TokenCredential {
       );
 
       const arrayScopes = Array.isArray(scopes) ? scopes : [scopes];
-      return this.msalFlow.getToken(arrayScopes, newOptions);
+      const certificate = await this.buildClientCertificate();
+      return this.msalClient.getTokenByClientCertificate(arrayScopes, certificate, newOptions);
     });
   }
+
+  private async buildClientCertificate(): Promise<CertificateParts> {
```

Comment:
Copied verbatim from [MsalClientCertificate](https://github.com/maorleger/azure-sdk-for-js/blob/main/sdk/identity/identity/src/msal/nodeFlows/msalClientCertificate.ts#L55-L56)